packages/core/src/external/aws/lambda-logic/bundle-to-html-adhd.ts

```diff
@@ -0,0 +1,2339 @@
+                <a href="#conditions">Conditions</a>
+              </li>
+              <li>
+                <a href="#allergies">Allergies</a>
+              </li>
+              <li>
+                <a href="#procedures"
+                  >Procedures</a
+                >
+              </li>
+              <li>
+                <a href="#social-history">Social History</a>
+              </li>
+              <li>
+                <a href="#vitals">Vitals</a>
+              </li>
+            </div>
+            <div class='half'>
+              <li>
+                <a href="#laboratory">Laboratory</a>
+              </li>
+              <li>
+                <a href="#other-observations">Other Observations</a>
+              </li>
+              <li>
+                <a href="#immunizations">Immunizations</a>
+              </li>
+              <li>
+                <a href="#family-member-history">Family Member History</a>
+              </li>
+              <li>
+                <a href="#related-persons">Related Persons</a>
+              </li>
+              <li>
+                <a href="#tasks">Tasks</a>
+              </li>
+              <li>
+                <a href="#coverage">Coverage</a>
+              </li>
+              <li>
+                <a href="#encounters">Encounters</a>
+              </li>
+            </div>
+            </ul>
+        </div>
+      </div>
+    </div>
+  `;
+}
+
+function createHeaderTableRow(label: string, value: string) {
+  return `
+    <tr>
+      <td>
+        <span class="header-label">${label}</span>
+        <span>
+          ${value}
+        </span>
+      </td>
+    </tr>
+  `;
+}
+
+type EncounterTypes =
+  | "labs"
+  | "progressNotes"
+  | "afterInstructions"
+  | "reasonForVisit"
+  | "documentation";
+
+type EncounterSection = {
+  [key: string]: {
+    [k in EncounterTypes]?: DiagnosticReport[];
+  };
+};
+
+function createDiagnosticReportsSection(
+  diagnosticReports: DiagnosticReport[],
+  practitioners: Practitioner[],
+  aweAndADHDVisits: Condition[],
+  organizations: Organization[]
+) {
+  const mappedPractitioners = mapResourceToId<Practitioner>(practitioners);
+  const mappedOrganizations = mapResourceToId<Organization>(organizations);
+
+  if (!diagnosticReports) {
+    return "";
+  }
+
+  const encounterSections: EncounterSection = buildEncounterSections({}, diagnosticReports);
+
+  const encountersWithoutAWEAndADHD = Object.entries(encounterSections)
+    .filter(([key]) => {
+      const encounterDateFormatted = dayjs(key).format(ISO_DATE);
+      const leftOverVisit = aweAndADHDVisits.find(visit => {
+        const visitDate = visit.onsetDateTime ?? "";
+        const visitDateFormatted = dayjs(visitDate).format(ISO_DATE);
+
+        return visitDateFormatted === encounterDateFormatted;
+      });
+
+      return !leftOverVisit;
+    })
+    .reduce((acc, [key, value]) => {
+      acc[key] = value;
+      return acc;
+    }, {} as EncounterSection);
```

Comment:
Now passing the filtered encounters to the `buildReports` functions to be able to reusue `buildReports`  for the adhd and awe sections